src/transformers/utils/fx.py

```diff
@@ -882,11 +882,46 @@ def call_module(self, m, forward, args, kwargs):
     def proxy(self, node):
         return HFProxy(node, self)
 
-    def trace(self, root: PreTrainedModel, concrete_args: Optional[Dict[str, Any]] = None) -> Graph:
+    def trace(
+        self,
+        root: Union[torch.nn.Module, Callable[..., Any]],
+        concrete_args: Optional[Dict[str, Any]] = None,
+        dummy_inputs: Optional[Dict[str, Any]] = None,
+        infer_concrete_args_from_dummy_inputs: bool = True,
+    ) -> Graph:
+        """
+        Traces `root` and returns the corresponding FX `torch.fx.Graph` representation. `root` can either be a
+        `torch.nn.Module` instance or a Python callable. Note that after this call, `self.root` may be different from
+        the `root` passed in here. For example, when a free function is passed to `trace()`, we will create a
+        `torch.nn.Module` instance to use as the root and add embedded constants to.
+
+        Args:
+            root (`torch.nn.Module` or  `Callable`):
+                Either a `torch.nn.Module`` or a function to be traced through. If root is not a
+                [`~transformers.PreTrainedModel`], then `dummy_inputs` must be passed, otherwise tracing will fail.
+            concrete_args (`Dict[str, Any], *optional*):
+                Concrete arguments that should not be treated as Proxies
+            dummy_inputs (`Dict[str, Any]`, *optional*):
+                The dummy inputs needed to handle data-dependent control-flow if `root` is not a
+                [`~transformers.PreTrainedModel`]. It can also be used when `root` is a
+                [`~transformers.PreTrainedModel`] to specify custom dummy inputs for a subset or all the model inputs.
+            infer_concrete_args_from_dummy_inputs (`bool`, *optional*, defaults to `True`):
```

Comment:
No, the argument is optional since it has a default (the user does not have to provide it). Please read the [writing documentation](https://github.com/huggingface/transformers/tree/main/docs#writing-documentation---specification) guide.

Reply:
I was unaware of this. I mistakenly thought that `bool, *optional*` means `Optional[bool]` in `typing` nomenclature. This seems weird to me as all defaults become optional now? Anyway perhaps a chat we can have someplace else.

Reply:
All arguments that have a default are optionals, yes. That is the definition of an optional argument, an argument you do not need to provide. It's not because the `typing` module decide to (badly) reuse that word for something else that this will change.